### PR TITLE
WIP add_cms_api

### PR DESCRIPTION
**Pull request by andrewdavidmackenzie**
A WIP Pull Request to add support in the 3scale ruby api library for CMS API

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,4 +8,5 @@ group :development do
   gem 'dotenv'
   gem 'pry-byebug', platform: :mri
   gem 'yard'
+  gem 'rspec'
```
Comment (eguzki) [inline Gemfile]: We tend to declare dependencies in `3scale-api.gemspec`.

`rspec` is declared there. 

It is true, though, there are some dependencies declared in Gemfile that should be moved to `3scale-api.gemspec`.

File: lib/3scale/api/client.rb [hunk 2]
```diff
@@ -816,6 +816,19 @@ def show_provider
...
+      # @api public
+      # @param [String] kind Kind of object to list in ['template', 'file', 'section']
+      # @param [Fixnum] page Page of results to return
+      # @return [Hash]
+      # NOTE: when listing 'templates' the return list can contain objects of type 'partial' and/or 'page'
+      # so the subkind parameter can be used to specify which to include in the list returned
+      def list_cms(kind, page, subkind=nil)
+        params = { per_page: 100, page: page}.reject { |_, value| value.nil? }
+        response = http_client.get("/admin/api/cms/#{kind}s", params: params)
+        subkind = "#{kind}" unless subkind
```
Comment (eguzki) [inline lib/3scale/api/client.rb]: [Rubocop](https://www.rubocop.org) linter complained in two ways:
* Prefer `to_s` over string interpolation
* Use `||=` operator instead

```
subkind ||= kind.to_s
```

File: Gemfile (hunk 1 above)
Comment (andrewdavidmackenzie) [reply Gemfile]: OK, I'll remove this. Want to move any of the others - it is spread a bit between the two files....???

File: lib/3scale/api/client.rb (hunk 2 above)
Comment (andrewdavidmackenzie) [reply lib/3scale/api/client.rb]: OK.

**Comment by eguzki** [comment]
@andrewdavidmackenzie you have been set in the `assignee` field. Update that field when you are done and want some further review

**Comment by andrewdavidmackenzie** [comment]
By sheer coincidence I came across this today... still open but out of date. Would be nice to add the CMS API stuff in here some day....but I wasn`t able to complete that part of toolbox...

**Comment by abarrak** [comment]
Hi @andrewdavidmackenzie,
A good addition of CMS APIs, looking forward for merging it,
There are minor conflicts currently though.

**Pull request closed without merging**